usr/share/rear/prep/GNU/Linux/210_include_dhclient.sh

```diff
@@ -4,114 +4,123 @@
 # copy the binaries and config files that we require to use dhclient/dhcpcd
 # on the rescue image (IPv4/IPv6)
 
-define_dhclients_variable()
-{   local x
-    dhclients=()
-    for x in "${DHCLIENT_BIN##*/}" \
-        "${DHCLIENT6_BIN##*/}" \
-        dhcpcd dhclient \
-        dhcp6c dhclient6 ;
-    do
-        [ "x$x" == x ] && continue
-        for d in ${dhclients[@]} ; do
-            [ "x$d" = "x$x" ] && continue 2
-        done
-        dhclients=(${dhclients[@]} "$x")
-    done
-    dhclients=${dhclients[@]}
-}
+# Respect an explicit rejection
+if is_false "$USE_DHCLIENT"; then
+    Log "Disabling DHCP client on the rescue system (USE_DHCLIENT='$USE_DHCLIENT')"
+    return 0
+fi
 
-dhcp_interfaces_active() {
-    local my_dhclients
-    my_dhclients=${dhclients}
-    my_dhclients=${dhclients// /|}
-    ps -e | grep -qEs "[ /]($my_dhclients)"
+# Array of known DHCP clients, which must also be known to skel/default/etc/scripts/system-setup.d/58-start-dhclient.sh
+local dhcp_clients=(dhcpcd dhclient dhcp6c dhclient6)
+
+# DHCLIENT_BIN, if set, specifies the DHCP client binary to use on the rescue system.
+# Must be one of the binaries listed in $dhcp_clients. Default (empty): try to auto-detect.
+# FIXME: DHCLIENT_BIN could be documented in default.conf as needed.
+if [[ -n "$DHCLIENT_BIN" ]]; then
+    has_binary "$DHCLIENT_BIN" || Error "DHCLIENT_BIN='$DHCLIENT_BIN' but such a binary could not be found"
+    [[ "${dhcp_clients[*]}" == *" $DHCLIENT_BIN "* ]] || Error "DHCLIENT_BIN='$DHCLIENT_BIN' is not among known DHCP clients (${dhcp_clients[*]})"
+fi
+
+
+function dhcp_client_is_active() {
+    # Strategy 1: Check if any DHCP client process is running (does not need to be the one specified by DHCLIENT_BIN)
+    local dhcp_clients_regexp="${dhcp_clients[*]}"
+    dhcp_clients_regexp=${dhcp_clients_regexp// /|}
+    ps -e | grep -qEs "[ /]($dhcp_clients_regexp)"
     if [ $? -eq 0 ]; then
```

Comment:
Is there a reason why not simpler and more straightforward
```
if ps -e | grep -qEs "[ /]($dhcp_clients_regexp)" ; then
```
?

Reply:
Yes, the reason is that there would be something for archeologists to discover! ;-)

(Changed the inherited lines.)